Elizabeth/elizabeth_practice.c

```diff
@@ -8,6 +9,183 @@ const int TEMP_SENSE_MAX = 1023;
 const float TEMP_C_IF_INVALID = -100.0;
 const float TEMP_F_IF_INVALID = -200.0;
 
+const int MAX_LINKED_LIST_LENGTH = 10;
+
+typedef struct Node
+{
+   int value;
+   struct Node *this_node_addr;
+   struct Node *next_node_addr;
+} Node;
+
+void add_ll_element(int value, Node *linked_list, int *filled_size);
+void print_ll(Node linked_list[], int *filled_size);
+void remove_all_nodes_with_value(int value_to_remove, Node linked_list[], int *filled_size);
+
+void add_ll_element(int value, Node *linked_list, int *filled_size)
+{
+   if (*filled_size == 0)
+   {
+      Node node_to_add;
+      node_to_add.value = value;
+      node_to_add.this_node_addr = linked_list + *filled_size;
+      node_to_add.next_node_addr = NULL;\
+
+      *(linked_list) = node_to_add;
+
+      *filled_size = *filled_size + 1;
+   }
+   else if (*filled_size < MAX_LINKED_LIST_LENGTH)
+   {
+      Node node_to_add;
+      Node node_to_update = *(linked_list + *filled_size - 1);
+
+      Node *next_addr_before = node_to_update.next_node_addr;
+
+      Node *next_addr_after = linked_list + *filled_size;
+
+      printf("last node next addr before: %p, changed to %p \n",
+             next_addr_before, next_addr_after);
+
+      node_to_add.value = value;
+      node_to_add.this_node_addr = next_addr_after;
+      node_to_add.next_node_addr = NULL;
+
+      *(linked_list + *filled_size) = node_to_add;
+
+      node_to_update.next_node_addr = next_addr_after;
+      *(linked_list + *filled_size - 1) = node_to_update;
+
+      *filled_size = *filled_size + 1;
+   }
+   else
+   {
+      printf("Linked List is full, not adding element\n");
+   }
+}
+
+void remove_all_nodes_with_value(int value_to_remove, Node linked_list[],
+                                 int *filled_size)
+{
+   if (filled_size > 0)
+   {
+      Node first_node = *(linked_list);
+      Node *current_node_addr;

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The scope of the variable 'current_node_addr' can be reduced.](https://app.codacy.com/manual/e-mock/CProgramming-Team6/pullRequest?prid=5603305)